packages/next/server/config-shared.ts

```diff
@@ -169,6 +169,10 @@ export type NextConfig = { [key: string]: any } & {
     urlImports?: NonNullable<webpack5.Configuration['experiments']>['buildHttp']
     outputFileTracingRoot?: string
     outputStandalone?: boolean
+    relay?: {
+      artifactDirectory?: string
+      language: 'typescript' | 'javascript'
+    }
```

Comment:
do you have to duplicate this in both relay.config and next.js config?

Can we figure out how to use a single `relay` config for this, to avoid duplication?

Reply:
Yeah I looked into this but the Relay Compiler crates aren't published anywhere right? I suppose if we published them we could use the crate to read the config files and determine the artifact directory for a given source file. 

Sharing logic here seems reasonable.

Reply:
I suppose we could assume it's a `SingleProjectConfig` and just read off of that. I only mentioned bringing in the crate because determining the artifact location is a bit more complex with a multi project config.

Reply:
> Yeah I looked into this but the Relay Compiler crates aren't published anywhere right? I suppose if we published them we could use the crate to read the config files and determine the artifact directory for a given source file.
> 
> Sharing logic here seems reasonable.

No, we didn't publish it, but I think you can still us it as a dependency: 

```
relay-compiler = { git = "https://github.com/facebook/relay" tag="v13.0.0" }
```

Reply:
Additionally, I think when this lands and if we can use the relay config from the `relay-compiler` we can add support for loading this configuration from next.js config file in the relay-compiler.

 Here: https://github.com/facebook/relay/blob/main/compiler/crates/relay-compiler/src/main.rs#L102

Reply:
Oh nice, didn't realize I could pull from a git repo + tag. I'll try this out. 

I don't think people would expect to have their relay config inside of their next config. I think we should keep the `relay.json` as the source of truth. Let me see what I can cook up.

Reply:
> I don't think people would expect to have their relay config inside of their next config. I think we should keep the relay.json as the source of truth. Let me see what I can cook up.

yeah, I just want to make sure people don't have to setup relay config twice (all these language/artifactsDirectory) in next.js and relay compiler. It would be great if next.js would just read all needed values from relay configuration. That's how it works now with babel: https://github.com/facebook/relay/blob/main/packages/babel-plugin-relay/BabelPluginRelay.js#L17-L49

Reply:
Is api for loading config for relay exported from the crate?
I think it will be more reliable, as this code is in 3rd party repository, unlike the babel plugin.